app/jobs/populate_register_data_in_db_job.rb

```diff
@@ -1,11 +1,22 @@
 class PopulateRegisterDataInDbJob < ApplicationJob
   queue_as :default
 
+  module Exceptions
+    class FrontendInvalidRegisterError < StandardError; end
+  end
+
   def perform(register)
     Delayed::Worker.logger.info("Updating #{register.name} in database")
+    begin
     # Initialize client and download / update data
-    register_client = @@registers_client.get_register(register.name.parameterize, register.register_phase.downcase, PostgresDataStore.new(register))
-    register_client.refresh_data
+      register_client = @@registers_client.get_register(register.name.parameterize, register.register_phase.downcase, PostgresDataStore.new(register))
+      register_client.refresh_data
+    rescue InvalidRegisterError
+      # If register data is invalid we want to delete existing entries and records to force a full reload
+      Record.where(register_id: register.id).destroy_all
+      Entry.where(register_id: register.id).destroy_all
```

Comment:
Do we want to control that both `destroy_all` are successful? Or, what happens if the first succeeds but the second fails? Is it a tolerable inconsistency?

Reply:
id personally use `find` here instead of `where`

Reply:
@danmitchell- I don't _think_ `find` works for this case, as we don't want to delete one item by ID. We want to destroy all matching results